### PR TITLE
`Development`: Fix an issue while parsing results

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
We saw a null pointer exception on production.

### Description
Added a null check.

### Steps for Testing
code review

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

**Review by sarpsahinalp — DISMISSED** [review]
Code lgtm

File: src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/TestResultXmlParser.java [hunk 1]
```diff
@@ -60,6 +64,12 @@ record TestSuites(@JacksonXmlElementWrapper(useWrapping = false) @JacksonXmlProp
 
     @JsonIgnoreProperties(ignoreUnknown = true)
     record TestSuite(@JacksonXmlElementWrapper(useWrapping = false) @JacksonXmlProperty(localName = "testcase") List<TestCase> testCases) {
+
+        TestSuite {
+            if (testCases == null) {
+                testCases = Collections.emptyList();
+            }
+        }
```
[the remark this answers is not in the document]
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/TestResultXmlParser.java]: I improved this in f4f6e655a9c519d6b9d93f47ad1904aec63290a8

**Review by laadvo — APPROVED** [review]
Code LGTM

**Review by pzdr7 — APPROVED** [review]
Code

**Review by SimonEntholzer — APPROVED** [review]
Code change makes sense :+1:

**Review by krusche — APPROVED** [review]
Code looks good 👍

**Pull request merged**